server/plugin/utils.go

```diff
@@ -454,7 +454,8 @@ func (p *Plugin) UpdatePipelineRunApprovalPost(approvalSteps []*serializers.Appr
 }
 
 func (p *Plugin) deleteSubscription(subscription *serializers.SubscriptionDetails, mattermostUserID string) (int, error) {
-	if statusCode, err := p.Client.DeleteSubscription(subscription.OrganizationName, subscription.SubscriptionID, mattermostUserID); err != nil {
+	// On deletion if a subscription is not found on the Azure DevOps portal then delete it from the Mattermost's KV store
+	if statusCode, err := p.Client.DeleteSubscription(subscription.OrganizationName, subscription.SubscriptionID, mattermostUserID); statusCode != http.StatusNotFound && err != nil {
```

Comment:
```suggestion
	if statusCode, err := p.Client.DeleteSubscription(subscription.OrganizationName, subscription.SubscriptionID, mattermostUserID); err != nil && statusCode != http.StatusNotFound {
```
Should we do this? Is there any case where status code is 404 and there could still be error?

Reply:
Don't think so, status 404 should have only 1 use-case that a resource is not found so irrespective of error if a subscription is not found on AZD then we should remove it from MM as well